broadcaster/backlog/backlog.go

```diff
@@ -350,26 +357,23 @@ func (s *backlogSegment) append(prevMsgIdx uint64, msg *m.BroadcastFeedMessage)
 // Contains confirms whether the segment contains a message with the given
 // sequence number.
 func (s *backlogSegment) Contains(i uint64) bool {
-	start := s.start.Load()
-	if i < start || i > s.end.Load() {
+	s.messagesLock.RLock()
+	defer s.messagesLock.RUnlock()
+	start := s.Start()
+	if i < start || i > s.End() {
```

Comment:
Same thing in this function about recursive locking

Reply:
Good catch! I have changed it over

Reply:
I think there's some calls to End that have the same issue